compiler/il/OMRBlock.cpp

```diff
@@ -1907,7 +1907,11 @@ OMR::Block::StandardException OMR::Block::_standardExceptions[] =
    {19, "ArithmeticException", CanCatchDivCheck },
    {19, "ArrayStoreException", CanCatchArrayStoreCheck },
    {19, "VirtualMachineError", CanCatchNew | CanCatchArrayNew },
+#if defined(J9VM_OPT_VALHALLA_VALUE_TYPES)
```

Comment:
This is an OpenJ9 ifdef in OMR code - if we are going to support compile-time enable/disable of value types then we are going to need an OMR macro to control this since we shouldn't be putting OpenJ9 macros or Java-specific macros into OMR which, by definition, is not language specific.

Reply:
At present the check is done in the compilation so I don't know that this can be ifdef'd and so we may need to select the right thing for the compile depending on the object model...

Reply:
In commit https://github.com/eclipse/omr/pull/5203/commits/71c19560f51e0846bbeeed2703fd861800a38c7d, added a second array of exceptions, [`_valueTypesExceptions`](https://github.com/eclipse/omr/blob/71c19560f51e0846bbeeed2703fd861800a38c7d/compiler/il/OMRBlock.cpp#L1920-L1924), that are included only if `OMR::ObjectModel::areValueTypesEnabled` returns `true`.

Also added an implementation of [`OMR::ObjectModel::areValueTypesEnabled`](https://github.com/eclipse/omr/blob/71c19560f51e0846bbeeed2703fd861800a38c7d/compiler/env/OMRObjectModel.hpp#L61).